File: greedy_decoding/greedy_decoding.go
# https://app.codesignal.com/challenge/JhtBKjkeDwaaERovM

func greedyDecoding(message string) (decodedMessage string) {    
    n := len(message)
    for i:=0; i < n; i++ {
        idxWidth := 1
        if i+1 < n {
            idxWidth = 2
        }
        chunk := message[i:i+idxWidth]
        chunkInt, _ := strconv.Atoi(chunk)
        
        if chunkInt <= 26 {
            decodedMessage += string(chunkInt+64)
            i++
        } else {
            smallChunk := message[i:i+1]
            smallChunkInt, _ := strconv.Atoi(smallChunk)
            decodedMessage += string(smallChunkInt+64)
        }
    }
    
    return
}

File: greedy_decoding/greedy_decoding.rb
# https://app.codesignal.com/challenge/JhtBKjkeDwaaERovM

def greedyDecoding(message)    
    i = 0
    decoded_msg = ""
    loop do
        break if i >= message.size
        l = message[i..i+1].to_i
        
        if l <= 26
            offset = l
            i+=2
        else
            offset = message[i].to_i
            i+=1
        end
        
        decoded_msg += (offset+64).chr
    end
    
    decoded_msg
end

